bip-blind-merged-mining.md

```diff
@@ -0,0 +1,329 @@
+
+Where does this data come from, and how does it get around?
+
+#### Creating / Broadcasting This Data
+
+##### Creation
+
+By the time Blind Merged Mining can take place, the ChainIndex is globally known (it is the "Account Number" in D1 [see previous BIP], and "nSidechain" in the code). Each sidechain, when activated by soft fork, will take one of the 0-255 available indexes.
+
+The other two items, sideHeaderHash and prevBlockRef, are created by sidechain nodes. sideHeaderHash is quite straightforward -- side:nodes build side:blocks, and take the hash of these.
+
+The final item, prevBlockRef, is a little more complicated. It is an integer that counts the number of "skips" one must take in the side:chain in order to find the current side:block's parent block. In practice, this value will usually be zero. It will only be a value other than zero, in cases where invalid sidechain blocks have been mined, or when a side:node intentionally wants to orphan some side:blocks (if a side:node wants to orphan the most-recent N blocks, the value of the current block will be equal to N ; in the block after that it will be back to zero).
+
+![dots-image](/bip-blind-merged-mining/bmm-dots-examples.png?raw=true)
+
+Since the hashes themselves are already ordered by the mainchain, tracing the blockchain's path by index (prevBlockRef) will be the same as tracing it by identifying a list of hashes. In other words, the ordering given via each side:block's "prevBlockRef" will be isomorphic to an ordering given by each side:block's "prevSideHeaderHash" ... if "prevSideHeaderHash is defined to be the sidechain's equivalent of the mainchain's "prevBlockHash". It will be possible to freely convert from one to the other. See M8 to learn more about how these hashes are requested by sidechain block creators to be included in the mainchain.
+
+Now that we know what our critical data is, and how it is made, how is this data broadcast and stored?
+
+##### Broadcast
+
+Mainchain nodes are going to need this data later, so it must be easy to find. We will put it into the coinbase via OP RETURN.
+
+#### M7 -- "Blind-Mine the Sidechain(s)"
+
+Thus, (for n sidechains) we have a coinbase output with:
+
+    1-byte - OP_RETURN (0x6a)
+    1-byte - Push the following (4+(n*35)) bytes (0x??)
+    4-byte - Message header (0xD3407053)
+    (n*(32+5))-byte - A sequence of bytes, of the three Mini-Header items (h*, prevBlockRef, ChainIndex).
+
+( We assume that 5 bytes are used for the Critical Data bytes (non h* parts of the Sidechain Mini-Header). For 256 sidechains, a total of 9,478 bytes [1+1+4+256\*(32+5)] are required, conveniently just below the 10 KB scriptPubKey size limit.)

```

Comment:
Why? This is useless data to the main chain.

Reply:
The mainchain needs this data in order to validate the two styles of "M8" transaction.